lib/test.js

```diff
@@ -225,29 +225,25 @@ Test.prototype._assert = function assert (ok, opts) {
             }
             
             var s = m[1].split(/\s+/);
-            var filem = /((?:\/|[A-Z]:\\)[^:\s]+:(\d+)(?::(\d+))?)/.exec(s[1]);
-            if (!filem) {
-                filem = /((?:\/|[A-Z]:\\)[^:\s]+:(\d+)(?::(\d+))?)/.exec(s[2]);
-                
-                if (!filem) {
-                    filem = /((?:\/|[A-Z]:\\)[^:\s]+:(\d+)(?::(\d+))?)/.exec(s[3]);
-
-                    if (!filem) {
-                        continue;
-                    }
-                }
+            var filemRe = /((?:\/|[A-Z]:\\)[^:\s]+:(\d+)(?::(\d+))?)/;
+            var filem;
+            var sIndex;
+            for (sIndex in s.slice(0, 4)) {
```

Comment:
for..in loops should be avoided; this would be better done as a for loop, or as `Array.prototype.some.call` or something

Reply:
`Array.prototype.forEach` would be the cleanest way to do this, but it's not supported in older versions of Node (or older browsers) without polyfill support. Not sure what Tape's polyfill policy is.

I can switch it to some variation of `for (sIndex = 0; sIndex < s.slice(0, 4).length; s++) { ... }` though.

(I'm not sure what expectations Tape makes

Reply:
True; we have the `for-each` package to handle `forEach` but that doesn't handle `some`.